### ISSUE TITLE
Dependabot does not seem to run on latest version

### ISSUE THREAD

**Issue by LucaGallinari** [dependabot/dependabot-core#5251]
Hi all, we are using Dependabot in various private repositories. We noticed some days ago that Dependabot updated one of our dependency to a new minor version as it does normally, BUT it also updated some other dependencies by using a PHP version that is not compatible with the one specified in the composer.json.
We tried to replicate locally the command that Dependabot has executed, and it seems to be this:

`composer require webgriffe/sylius-upgrade-plugin "^0.5.0" -W --dev --ignore-platform-reqs`

We dind't expect it to specify the "--ignore-platform-reqs" argument, cause the latest version of dependabot-core does this (you can find all references to this code [here](https://github.com/dependabot/dependabot-core/search?q=setIgnorePlatformRequirements)):

`->setIgnorePlatformRequirements(false);`

It seems like this should tell Dependabot to not specify the "--ignore-platform-reqs" parameter.
[Here you can see the commit](https://github.com/dependabot/dependabot-core/commit/7f7327bf68dbd4fed05e6e4ea5a1f3b47a01089f) that added that line of code and it was done in late 2020. So, we assumed (until now) that the Dependabot version that opens PRs on our repositories was the latest, and so it should have that change.

Moreover, we found out, in a really old version of Dependabot, that this parameter depended on the Composer "platform" config: https://github.com/dependabot/dependabot-core/pull/527/files#diff-c53ba485e227c954aa17213418a456cbf68e38eb9ca6efe654f61937e987f967R75
And this seems to be what Dependabot is doing on our repositories...

First of all, how can we make Dependabot run with the latest version?
Secondly, it would be great if every PR included the "update/require" command executed by Dependabot. It would help a lot. Did you ever though about this? I can open another issue for this maybe.

**Package ecosystem**
Composer
**Package manager version**
2.*
**Language version**
PHP 8.0

**Comment by mattt** [comment]
Thanks for taking the time to describe this behavior, @LucaGallinari. To clarify, are you specifying a PHP `platform` version in your manifest?

> It seems like this should tell Dependabot to not specify the "--ignore-platform-reqs" parameter.
> [Here you can see the commit](https://github.com/dependabot/dependabot-core/commit/7f7327bf68dbd4fed05e6e4ea5a1f3b47a01089f) that added that line of code and it was done in late 2020.

That commit changes the behavior of our helper for Composer v1. Here's the same (most recent) change for v2: https://github.com/dependabot/dependabot-core/commit/4c0eb1296748d44df5de4fcefd4b326c7fe8c347, which corresponds to the PR adding support for Composer v2 (https://github.com/dependabot/dependabot-core/pull/2838). Does that PR give you any context for what's going on?

> [I]t would be great if every PR included the "update/require" command executed by Dependabot. It would help a lot. Did you ever though about this? I can open another issue for this maybe.

I think that's a great idea, @LucaGallinari. If you have a moment, please do write that up.

**Comment by LucaGallinari** [comment]
Any update on this?

**Comment by jurre** [comment]
No update, but I have some clarifying questions.

So if I understand correctly, you are saying Dependabot should treat `platform.php` the same as `require.php`, both resulting in the same PHP constraint being applied, and `--ignore-platform-reqs` not being used? What if when both are specified and they don't match, does one always win or do we use the lowest version?

Currently Dependabot relies on having just a single PHP version installed, which is nice because it means we don't have to install the version on the fly. Would either of these approaches require us installing that specific version of the runtime?

Would you be able to work around this by using the `platform.php` config option?

Hope I didn't misunderstand anything, if I did please let me know!

In an attempt to be transparent, we update a lot of PHP projects but hadn't heard of this issue before, so it might be a while before we can get to fixing it. I'd be happy to review and land a PR that fixes it though!

**Comment by LucaGallinari** [comment]
Hi @jurre,
thanks for answering. Btw the problem is not that complex as you described.

We can focus on this one thing: dependabot MUST **NEVER** EVER use the "--ignore-platform-reqs" parameter when updating dependencies but it seems like it does (i described why this is bad in the first half of my [previous comment](https://github.com/dependabot/dependabot-core/issues/5251#issuecomment-1155332066)).

The **actual** version of the Updater class for Composer v2 [is this](https://github.com/dependabot/dependabot-core/blob/main/composer/helpers/v2/src/Updater.php#L88) and it does this:

```->setIgnorePlatformRequirements(false);```

and this would be good, but instead it seems to do this **actually**:

```
if ($config->get('platform') === []) {
    $install->setIgnorePlatformRequirements(true);
}
```

that we found out was an old version of the Updater for Composer, [here you can see the commit](https://github.com/dependabot/dependabot-core/pull/527/files#diff-c53ba485e227c954aa17213418a456cbf68e38eb9ca6efe654f61937e987f967R75). I described this behavior in the second half of my [previous comment](https://github.com/dependabot/dependabot-core/issues/5251#issuecomment-1155332066).

It seems like Dependabot is not running on the latest version. How can we debug this?

**Comment by jurre** [comment]
Thanks!

> It seems like Dependabot is not running on the latest version. How can we debug this?

Pretty sure it is running the latest version, so there's probably something else going on here. You can verify this by looking at your logs (`github.com/<your>/<repo>/network/updates`, click on `Last checked ...` for the Composer entry there), it'll say something along the lines of:

```
updater_timeout=45m0s updater_version=0.202.0-15248069ef812213941be5bf0c0f0659b8c7fa7a
```

That `updater_version` refers to the version of dependabot-core that is used.

I'm not saying Dependabot is doing the right thing, I just don't think that we're running a (very) old version of the updater in production and that something else is causing the wrong behavior here.

One thing that _might_ be causing this, is that for some reason Dependabot seems to think you're still on composer v1, and it's using [this code](https://github.com/dependabot/dependabot-core/blob/24f9ee3a9925d453c034791f2796b38218224a36/composer/helpers/v1/src/Updater.php#L86) for the composer v1 updater.

To determine the composer version, Dependabot mainly looks to see if there's a `plugin-api-version: 1.x.x` in the lockfile, although the actual code it runs checks for a few more things: https://github.com/dependabot/dependabot-core/blob/24f9ee3a9925d453c034791f2796b38218224a36/composer/lib/dependabot/composer/helpers.rb#L16-L23

Could this be causing the issue?

**Comment by LucaGallinari** [comment]
Oh great, i'll check!

**Comment by LucaGallinari** [comment]
Ok. I confirm that the "project" is using the latest version of dependabot.
I dived in the source code of this repo, especially the part that uses the Composer library and found out that it's possible to run and debug locally an update of the dependencies through dependabot. This way, i was able to reproduce the same "wrong" behavior that we are experiencing.
I have to do some tests and i'll be back.

**Comment by jeffwidman** [comment]
@LucaGallinari any update on this?

**Comment by LucaGallinari** [comment]
Hi @jeffwidman and @jurre , sorry for the late response but i waited a bit as i wanted to be sure that what we "did" solved the issue by checking if Dependabot tried to open some other "problematic PRs", but it didn't.

So, it's not a Depedanbot issue! In a previous comment i wrote this:

> We did some with tests with Dependabot with different composer.json configs, and all these versions had the php version "~8.0.0" under the "require" node, and the results are these:
> - If we specify the version of PHP (like "8.0.19") even under the "platform" config, Dependabot will consider the constraint on the PHP version, thus it's updating dependencies without specifying the "--ignore-platform-reqs" param.
> - If we DON'T specify the PHP version under "platform", it will ignore the "php" version specified under the "php" require node, thus it will update dependencies with "--ignore-platform-reqs" param.

But this seems to be false, probably we messed up some tests or we had something cached, because with "~8.0.0" as required version for PHP it properly works: Dependabot do not try to update packages by using a different minor version of PHP (like 8.1, 8.2 ecc), but it sticks to 8.0.x. I tried to simulate this by running Dependabot locally and it works properly.

I am sorry for the trouble...
Thank you very much! This tool is really usefull

**Comment by jeffwidman** [comment]
Great! Thanks for following up to let us know.

**Comment by eliashaeussler** [comment]
Hi together, sorry for coming back to this issue, but we have a similar issue with https://github.com/CPS-IT/project-builder (and some more repositories).

I was able to reproduce this by running Dependabot locally.

<details>
<summary>Logs of a local test run</summary>

```
[dependabot-core-dev] ~/dependabot-core $ DEBUG_HELPERS=true bin/dry-run.rb composer cps-it/project-builder
=> fetching dependency files
=> dumping fetched dependency files: ./dry-run/cps-it/project-builder/
=> parsing dependency files
=> updating 34 dependencies: cocur/slugify, cuyz/valinor, guzzlehttp/guzzle, nyholm/psr7, opis/json-schema, psr/http-client, psr/http-factory, psr/http-message, sebastianfeldmann/cli, symfony/config, symfony/console, symfony/dependency-injection, symfony/event-dispatcher, symfony/expression-language, symfony/filesystem, symfony/finder, symfony/yaml, twig/twig, webmozart/assert, armin/editorconfig-cli, composer/composer, composer/xdebug-handler, donatj/mock-webserver, ergebnis/composer-normalize, friendsofphp/php-cs-fixer, phpstan/extension-installer, phpstan/phpstan, phpstan/phpstan-phpunit, phpstan/phpstan-strict-rules, phpstan/phpstan-symfony, phpstan/phpstan-webmozart-assert, phpunit/phpunit, rector/rector, seld/jsonlint

=== cocur/slugify (4.3.0)
 => checking for updates 1/34
🌍 --> GET https://repo.packagist.org/p2/cocur/slugify.json
🌍 <-- 200 https://repo.packagist.org/p2/cocur/slugify.json
 => latest available version is 4.3.0
    (no update needed as it's already up-to-date)

=== cuyz/valinor (1.2.0)
 => checking for updates 2/34
🌍 --> GET https://repo.packagist.org/p2/cuyz/valinor.json
🌍 <-- 200 https://repo.packagist.org/p2/cuyz/valinor.json
 => latest available version is 1.2.0
    (no update needed as it's already up-to-date)

=== guzzlehttp/guzzle (7.5.0)
 => checking for updates 3/34
🌍 --> GET https://repo.packagist.org/p2/guzzlehttp/guzzle.json
🌍 <-- 200 https://repo.packagist.org/p2/guzzlehttp/guzzle.json
 => latest available version is 7.5.0
    (no update needed as it's already up-to-date)

=== nyholm/psr7 (1.5.1)
 => checking for updates 4/34
🌍 --> GET https://repo.packagist.org/p2/nyholm/psr7.json
🌍 <-- 200 https://repo.packagist.org/p2/nyholm/psr7.json
 => latest available version is 1.5.1
    (no update needed as it's already up-to-date)

=== opis/json-schema (2.3.0)
 => checking for updates 5/34
🌍 --> GET https://repo.packagist.org/p2/opis/json-schema.json
🌍 <-- 200 https://repo.packagist.org/p2/opis/json-schema.json
 => latest available version is 2.3.0
    (no update needed as it's already up-to-date)

=== psr/http-client (1.0.1)
 => checking for updates 6/34
🌍 --> GET https://repo.packagist.org/p2/psr/http-client.json
🌍 <-- 200 https://repo.packagist.org/p2/psr/http-client.json
 => latest available version is 1.0.1
    (no update needed as it's already up-to-date)

=== psr/http-factory (1.0.1)
 => checking for updates 7/34
🌍 --> GET https://repo.packagist.org/p2/psr/http-factory.json
🌍 <-- 200 https://repo.packagist.org/p2/psr/http-factory.json
 => latest available version is 1.0.1
    (no update needed as it's already up-to-date)

=== psr/http-message (1.0.1)
 => checking for updates 8/34
🌍 --> GET https://repo.packagist.org/p2/psr/http-message.json
🌍 <-- 200 https://repo.packagist.org/p2/psr/http-message.json
 => latest available version is 1.0.1
    (no update needed as it's already up-to-date)

=== sebastianfeldmann/cli (3.4.1)
 => checking for updates 9/34
🌍 --> GET https://repo.packagist.org/p2/sebastianfeldmann/cli.json
🌍 <-- 200 https://repo.packagist.org/p2/sebastianfeldmann/cli.json
 => latest available version is 3.4.1
    (no update needed as it's already up-to-date)

=== symfony/config (6.2.0)
 => checking for updates 10/34
🌍 --> GET https://repo.packagist.org/p2/symfony/config.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/config.json
 => latest available version is 6.2.0
    (no update needed as it's already up-to-date)

=== symfony/console (6.2.3)
 => checking for updates 11/34
🌍 --> GET https://repo.packagist.org/p2/symfony/console.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/console.json
 => latest available version is 6.2.3
    (no update needed as it's already up-to-date)

=== symfony/dependency-injection (6.2.3)
 => checking for updates 12/34
🌍 --> GET https://repo.packagist.org/p2/symfony/dependency-injection.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/dependency-injection.json
 => latest available version is 6.2.3
    (no update needed as it's already up-to-date)

=== symfony/event-dispatcher (6.2.2)
 => checking for updates 13/34
🌍 --> GET https://repo.packagist.org/p2/symfony/event-dispatcher.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/event-dispatcher.json
 => latest available version is 6.2.2
    (no update needed as it's already up-to-date)

=== symfony/expression-language (6.2.2)
 => checking for updates 14/34
🌍 --> GET https://repo.packagist.org/p2/symfony/expression-language.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/expression-language.json
 => latest available version is 6.2.2
    (no update needed as it's already up-to-date)

=== symfony/filesystem (6.2.0)
 => checking for updates 15/34
🌍 --> GET https://repo.packagist.org/p2/symfony/filesystem.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/filesystem.json
 => latest available version is 6.2.0
    (no update needed as it's already up-to-date)

=== symfony/finder (6.2.3)
 => checking for updates 16/34
🌍 --> GET https://repo.packagist.org/p2/symfony/finder.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/finder.json
 => latest available version is 6.2.3
    (no update needed as it's already up-to-date)

=== symfony/yaml (6.2.2)
 => checking for updates 17/34
🌍 --> GET https://repo.packagist.org/p2/symfony/yaml.json
🌍 <-- 200 https://repo.packagist.org/p2/symfony/yaml.json
 => latest available version is 6.2.2
    (no update needed as it's already up-to-date)

=== twig/twig (3.5.0)
 => checking for updates 18/34
🌍 --> GET https://repo.packagist.org/p2/twig/twig.json
🌍 <-- 200 https://repo.packagist.org/p2/twig/twig.json
 => latest available version is 3.5.0
    (no update needed as it's already up-to-date)

=== webmozart/assert (1.11.0)
 => checking for updates 19/34
🌍 --> GET https://repo.packagist.org/p2/webmozart/assert.json
🌍 <-- 200 https://repo.packagist.org/p2/webmozart/assert.json
 => latest available version is 1.11.0
    (no update needed as it's already up-to-date)

=== armin/editorconfig-cli (1.5.2)
 => checking for updates 20/34
🌍 --> GET https://repo.packagist.org/p2/armin/editorconfig-cli.json
🌍 <-- 200 https://repo.packagist.org/p2/armin/editorconfig-cli.json
 => latest available version is 1.5.2
    (no update needed as it's already up-to-date)

=== composer/composer (2.5.1)
 => checking for updates 21/34
🌍 --> GET https://repo.packagist.org/p2/composer/composer.json
🌍 <-- 200 https://repo.packagist.org/p2/composer/composer.json
 => latest available version is 2.5.1
    (no update needed as it's already up-to-date)

=== composer/xdebug-handler (3.0.3)
 => checking for updates 22/34
🌍 --> GET https://repo.packagist.org/p2/composer/xdebug-handler.json
🌍 <-- 200 https://repo.packagist.org/p2/composer/xdebug-handler.json
 => latest available version is 3.0.3
    (no update needed as it's already up-to-date)

=== donatj/mock-webserver (2.6.1)
 => checking for updates 23/34
🌍 --> GET https://repo.packagist.org/p2/donatj/mock-webserver.json
🌍 <-- 200 https://repo.packagist.org/p2/donatj/mock-webserver.json
 => latest available version is 2.6.1
    (no update needed as it's already up-to-date)

=== ergebnis/composer-normalize (2.29.0)
 => checking for updates 24/34
🌍 --> GET https://repo.packagist.org/p2/ergebnis/composer-normalize.json
🌍 <-- 200 https://repo.packagist.org/p2/ergebnis/composer-normalize.json
 => latest available version is 2.29.0
    (no update needed as it's already up-to-date)

=== friendsofphp/php-cs-fixer (3.13.2)
 => checking for updates 25/34
🌍 --> GET https://repo.packagist.org/p2/friendsofphp/php-cs-fixer.json
🌍 <-- 200 https://repo.packagist.org/p2/friendsofphp/php-cs-fixer.json
 => latest available version is 3.13.2
    (no update needed as it's already up-to-date)

=== phpstan/extension-installer (1.2.0)
 => checking for updates 26/34
🌍 --> GET https://repo.packagist.org/p2/phpstan/extension-installer.json
🌍 <-- 200 https://repo.packagist.org/p2/phpstan/extension-installer.json
 => latest available version is 1.2.0
    (no update needed as it's already up-to-date)

=== phpstan/phpstan (1.9.8)
 => checking for updates 27/34
🌍 --> GET https://repo.packagist.org/p2/phpstan/phpstan.json
🌍 <-- 200 https://repo.packagist.org/p2/phpstan/phpstan.json
 => latest available version is 1.9.14
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => latest allowed version is 1.9.8
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => requirements to unlock: update_not_possible
 => requirements update strategy: widen_ranges
    (no update possible 🙅‍♀️)

=== phpstan/phpstan-phpunit (1.3.3)
 => checking for updates 28/34
🌍 --> GET https://repo.packagist.org/p2/phpstan/phpstan-phpunit.json
🌍 <-- 200 https://repo.packagist.org/p2/phpstan/phpstan-phpunit.json
 => latest available version is 1.3.3
    (no update needed as it's already up-to-date)

=== phpstan/phpstan-strict-rules (1.4.4)
 => checking for updates 29/34
🌍 --> GET https://repo.packagist.org/p2/phpstan/phpstan-strict-rules.json
🌍 <-- 200 https://repo.packagist.org/p2/phpstan/phpstan-strict-rules.json
 => latest available version is 1.4.5
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => latest allowed version is 1.4.4
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => requirements to unlock: update_not_possible
 => requirements update strategy: widen_ranges
    (no update possible 🙅‍♀️)

=== phpstan/phpstan-symfony (1.2.19)
 => checking for updates 30/34
🌍 --> GET https://repo.packagist.org/p2/phpstan/phpstan-symfony.json
🌍 <-- 200 https://repo.packagist.org/p2/phpstan/phpstan-symfony.json
 => latest available version is 1.2.20
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => latest allowed version is 1.2.19
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => requirements to unlock: update_not_possible
 => requirements update strategy: widen_ranges
    (no update possible 🙅‍♀️)

=== phpstan/phpstan-webmozart-assert (1.2.2)
 => checking for updates 31/34
🌍 --> GET https://repo.packagist.org/p2/phpstan/phpstan-webmozart-assert.json
🌍 <-- 200 https://repo.packagist.org/p2/phpstan/phpstan-webmozart-assert.json
 => latest available version is 1.2.2
    (no update needed as it's already up-to-date)

=== phpunit/phpunit (9.5.27)
 => checking for updates 32/34
🌍 --> GET https://repo.packagist.org/p2/phpunit/phpunit.json
🌍 <-- 200 https://repo.packagist.org/p2/phpunit/phpunit.json
 => latest available version is 9.5.28
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => latest allowed version is 9.5.27
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => requirements to unlock: update_not_possible
 => requirements update strategy: widen_ranges
    (no update possible 🙅‍♀️)

=== rector/rector (0.15.4)
 => checking for updates 33/34
🌍 --> GET https://repo.packagist.org/p2/rector/rector.json
🌍 <-- 200 https://repo.packagist.org/p2/rector/rector.json
 => latest available version is 0.15.10
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => latest allowed version is 0.15.4
php -d memory_limit=-1 /opt/composer/v2/bin/run
{"error":"Your requirements could not be resolved to an installable set of packages.\n  Problem 1\n    - symfony\/config is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 2\n    - symfony\/console is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/console v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 3\n    - symfony\/dependency-injection is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/dependency-injection v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 4\n    - symfony\/event-dispatcher is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/event-dispatcher v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 5\n    - symfony\/expression-language is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/expression-language v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 6\n    - symfony\/filesystem is locked to version v6.2.0 and an update of this package was not requested.\n    - symfony\/filesystem v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 7\n    - symfony\/finder is locked to version v6.2.3 and an update of this package was not requested.\n    - symfony\/finder v6.2.3 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 8\n    - symfony\/yaml is locked to version v6.2.2 and an update of this package was not requested.\n    - symfony\/yaml v6.2.2 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n  Problem 9\n    - symfony\/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.\n    - guzzlehttp\/guzzle 7.5.0 requires symfony\/deprecation-contracts ^2.2 || ^3.0 -> satisfiable by symfony\/deprecation-contracts[v3.2.0].\n    - guzzlehttp\/guzzle is locked to version 7.5.0 and an update of this package was not requested.\n"}
PHP Deprecated:  Installer::setIgnorePlatformRequirements is deprecated since Composer 2.2, use setPlatformRequirementFilter instead. in /opt/composer/v2/vendor/composer/composer/src/Composer/Installer.php on line 1289
 => requirements to unlock: update_not_possible
 => requirements update strategy: widen_ranges
    (no update possible 🙅‍♀️)

=== seld/jsonlint (1.9.0)
 => checking for updates 34/34
🌍 --> GET https://repo.packagist.org/p2/seld/jsonlint.json
🌍 <-- 200 https://repo.packagist.org/p2/seld/jsonlint.json
 => latest available version is 1.9.0
    (no update needed as it's already up-to-date)
🌍 Total requests made: '34'
```

</details>

One of the relevant error messages is:

```
Your requirements could not be resolved to an installable set of packages.
  Problem 1
    - symfony/config is locked to version v6.2.0 and an update of this package was not requested.
    - symfony/config v6.2.0 requires php >=8.1 -> your php version (8.0.0; overridden via config.platform, actual: 7.4.33) does not satisfy that requirement.
```

Our library supports PHP 8.0 - 8.2 (that's probably the reason why [Dependabot configures a `platform.php` version `8.0.0` at runtime](https://github.com/dependabot/dependabot-core/blob/c8f6027606196d929baaac20ad66deffad419dc1/composer/lib/dependabot/composer/update_checker/version_resolver.rb#L162)). However, the locked version of `symfony/config` does not support PHP 8.0, neither does the version to be updated.

In this case, the update would have been possible if Dependabot had been executed with PHP 8.2. Reading through this issue, it seems like the only solution at the moment is to specify an updatable PHP version as `config.platform` in our `composer.json`. Is that correct? If yes, are there any plans to change this behavior? Or, at least, provide some configuration mechanism for such cases?

**Comment by jeffwidman** [comment]
Many thanks @LucaGallinari !
